From 2c01e67a74ca48ca87e2f11f6239d54c35551e7b Mon Sep 17 00:00:00 2001
From: David Adam <zanchey@ucc.gu.uwa.edu.au>
Date: Sat, 5 Jan 2019 15:59:25 +0800
Subject: [PATCH] histfile tests: tweak expect commands to avoid crash on
 32-bit platforms

Rather than killing the process with close, read EOF after sending the
"exit" command and wait for OS cleanup (per the expect examples).

Not cleaning up with wait caused expect to crash on all 32-bit platforms
including i586 and armv7l with "alloc: invalid block: 0xbf993ccb: 3d 3b".

64-bit platforms were not affected, for reasons that are not clear.
---
 tests/histfile.expect | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/histfile.expect b/tests/histfile.expect
index 81ca11276..7ac056235 100644
--- a/tests/histfile.expect
+++ b/tests/histfile.expect
@@ -92,7 +92,8 @@ expect_prompt -re "\r\n$hist_line\r\n" {
 # =============
 # Start by shutting down the previous shell.
 send "exit\r"
-close $spawn_id
+expect eof
+wait
 
 # Set the fish_history env var.
 set ::env(fish_history) env
